### PR TITLE
[feature] add DiscreteObject interface

### PR DESCRIPTION
**Pull request by smagnaniuni**
Simulation interface for discrete time updates.

How to use the TemporalAmount object:

If you have a rate of change in seconds, this is how you get the seconds from the argument, TemporalAmount does all the conversions, it doesn't matter if it's 1 day, 1 hour or 1 second.
```java
final TemporalAmount dt = Duration.ofHours(1);
System.out.println(dt.get(ChronoUnit.SECONDS)); // 3600
```

### CODE REVIEW

**Comment by smagnaniuni** [comment]
The Clock interface has an updateTick itself at the moment, I should remove it and extend this interface from the Clock's one.

File: app/src/main/java/homer/core/DiscreteObject.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+ * This interface models an object which is updated in discrete time steps.
+ */
+public interface DiscreteObject {
+
+    /**
+     * Sends the time update event to the object.
+     * 
+     * @param dt The time difference from the last tick.
+     */
+    void updateTick(TemporalAmount dt);
```
[the remark this answers is not in the document]
Comment (aleemont1) [reply app/src/main/java/homer/core/DiscreteObject.java:15]: It's the same to me.

**Review by aleemont1 — APPROVED** [review]

**Comment by smagnaniuni** [comment]
A `TemporalAmount` can take both a Duration or a Period.
A `Duration` represents an amount of time, you can get seconds out of it.
A `Period` represents the classic date format, you can get years, months and days out of it.
Maybe the `TemporalUnit` type should be replaced with `Duration` in order not to get confused.
You can get a list of supported units with `dt.getUnits()`
If a time derivative has days as the time unit, the deltaTime should first be converted to seconds, and then divided by the seconds in a day, or the derivative should be converted to seconds as the time unit.

File: app/src/main/java/homer/core/DiscreteObject.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (michelececcacci) [reply app/src/main/java/homer/core/DiscreteObject.java:15]: deltatime is more descriptive to me.

**Comment by michelececcacci** [comment]
LGTM

**Pull request merged**